### PR TITLE
Add passing_users to course summary

### PR DESCRIPTION
**Pull request by efischer19**
# [EDU-153](https://openedx.atlassian.net/browse/EDUCATOR-153)

This feels too easy - am I missing something?

### CODE REVIEW

**Comment by dsjen** [comment]
I just realized that we probably want to ensure that the `exclude` field can be used to exclude the `passing_users` field from the payload in both the top level and in the enrollment modes, since this functionality will be turned on later.  Please let me know if that doesn't make sense!

**Comment by efischer19** [comment]
No problem, I'll get that added this afternoon

File: analytics_data_api/management/commands/generate_fake_course_data.py [hunk 1]
```diff
@@ -161,15 +161,17 @@ def generate_daily_data(self, course_id, start_date, end_date):
             progress.update(1)
             date = date + datetime.timedelta(days=1)
 
-        for mode, ratio in enrollment_mode_ratios.iteritems():
+        for index, (mode, ratio) in enumerate(enrollment_mode_ratios.iteritems()):
             count = int(ratio * daily_total)
+            pass_rate = min(random.normalvariate(.45 + (.1 * index), .15), 1.0)
```
Comment (efischer19) [inline analytics_data_api/management/commands/generate_fake_course_data.py]: I utilize `index` here so we can get a higher pass rate for `verified` than we have for `audit` (scales with position in `enrollment_mode_ratios); is that overly complex for this data?

I also just guessed at the baseline pass rates (from 45% for audit up to 85% for verified), LMK if there's a more accurate value we ought to use.

File: analytics_data_api/v0/tests/views/test_course_summaries.py [hunk 2]
```diff
@@ -159,3 +163,10 @@ def test_programs(self):
         response = self.authenticated_get(self.path(exclude=self.always_exclude[:1], programs=['True']))
         self.assertEquals(response.status_code, 200)
         self.assertItemsEqual(response.data, self.all_expected_results(programs=True))
+
+    @ddt.data('passing_users', )
+    def test_exclude(self, field):
+        self.generate_data()
+        response = self.authenticated_get(self.path(exclude=[field]))
+        self.assertEquals(response.status_code, 200)
+        self.assertEquals(str(response.data).count(field), 0)
```
Comment (efischer19) [inline analytics_data_api/v0/tests/views/test_course_summaries.py]: `self.assertNotIn(field, response.data)` only works for the top-level (cumulative) field, not the per-mode field. This ensures the field is not present anywhere.

File: analytics_data_api/management/commands/generate_fake_course_data.py (hunk 1 above)
Comment (dsjen) [inline analytics_data_api/management/commands/generate_fake_course_data.py]: ❤️

File: analytics_data_api/v0/tests/views/test_course_summaries.py (hunk 2 above)
Comment (dsjen) [inline analytics_data_api/v0/tests/views/test_course_summaries.py]: Nice!

File: analytics_data_api/v0/views/course_summaries.py [hunk 3]
```diff
@@ -120,6 +121,10 @@ def postprocess_field_dict(self, field_dict):
             # don't do expensive looping for programs if we are just going to throw it away
             field_dict = self.add_programs(field_dict)
 
+        for field in self.exclude:
```
Comment (dsjen) [inline analytics_data_api/v0/views/course_summaries.py]: Thanks for taking care of that!

**Pull request merged**